instana/instrumentation/sanic_inst.py

```diff
@@ -0,0 +1,138 @@
+            span = async_tracer.active_span
+            if span is None:
+                wrapped(*args, **kwargs)
+            else:
+                response_details(span=span, response=response)
+                try:
+                    wrapped(*args, **kwargs)
+                except Exception as exc:
+                    span.log_exception(exc)
+                    raise
+
+
+        @wrapt.patch_function_wrapper('sanic.server', 'HttpProtocol.stream_response')
+        async def stream_with_instana(wrapped, instance, args, kwargs):
+            response = args[0]
+            span = async_tracer.active_span
+            if span is None:
+                await wrapped(*args, **kwargs)
+            else:
+                response_details(span=span, response=response)
+                try:
+                    await wrapped(*args, **kwargs)
+                except Exception as exc:
+                    span.log_exception(exc)
+                    raise
+
+
+    @wrapt.patch_function_wrapper('sanic.app', 'Sanic.handle_request')
+    async def handle_request_with_instana(wrapped, instance, args, kwargs):
+
+        try:
+            request = args[0]
+            try:  # scheme attribute is calculated in the sanic handle_request method for v19, not yet present
+                if "http" not in request.scheme:
+                    return await wrapped(*args, **kwargs)
+            except AttributeError:
+                pass
+            headers = request.headers.copy()
+            ctx = async_tracer.extract(opentracing.Format.HTTP_HEADERS, headers)
+            with async_tracer.start_active_span("asgi", child_of=ctx) as tracing_scope:
+                tracing_scope.span.set_tag('span.kind', 'entry')
+                tracing_scope.span.set_tag('http.path', request.path)
+                tracing_scope.span.set_tag('http.method', request.method)
+                tracing_scope.span.set_tag('http.host', request.host)
+
+                query = request.query_string
+
+                if isinstance(query, (str, bytes)) and len(query):
+                    if isinstance(query, bytes):
+                        query = query.decode('utf-8')
+                    scrubbed_params = strip_secrets_from_query(query, agent.options.secrets_matcher,
+                                                               agent.options.secrets_list)
+                    tracing_scope.span.set_tag("http.params", scrubbed_params)
+
+                if agent.options.extra_http_headers is not None:
+                    extract_custom_headers(tracing_scope, headers)
+                await wrapped(*args, **kwargs)
+                tracing_scope.span.set_tag("http.url", request.url)
+                if hasattr(request, "uri_template"):
+                    tracing_scope.span.set_tag("http.path_tpl", request.uri_template)
+                if hasattr(request, "ctx"):  # ctx attribute added in the latest v19 versions
+                    request.ctx.iscope = tracing_scope
```

Comment:
Can we put them together around L106-109?

Reply:
`"uri_template"` is only getting evaluated after the `await wrapped(*args, **kwargs)` line 122.
Line 123 I can move up. Lines 126-127 could go up, but it can be tricky for early v19 versions, as adding more things afterwards to the scope will require and update or inplace update of the request attribute if ctx exists. Additionally attaching the scope metrics to the request before executing the `await wrapped(*args, **kwargs)` exposes it to the customer, who may also use the ctx attribute for his own purpose and we should avoid him potentially cleaning the existing data. Do we have a reason for doing this? I suppose, if we want to give our already calculated tracing information to the customer for him to interact programatically, we could, but we should provide the customer with documentation of what is there and how to handle it.